server/polar/event/system.py

```diff
@@ -157,6 +163,38 @@ class CustomerDeletedEvent(Event):
         user_metadata: Mapped[CustomerDeletedMetadata]  # type: ignore[assignment]
 
 
+class SubscriptionCreatedMetadata(TypedDict):
+    subscription_id: str
+    product_id: str
+    amount: int
+    currency: str
+    recurring_interval: str
+    recurring_interval_count: int
+    started_at: str
+
+
+class SubscriptionCreatedEvent(Event):
+    if TYPE_CHECKING:
+        source: Mapped[Literal[EventSource.system]]
+        name: Mapped[Literal[SystemEvent.subscription_created]]
+        user_metadata: Mapped[SubscriptionCreatedMetadata]  # type: ignore[assignment]
+
+
+class SubscriptionCanceledMetadata(TypedDict):
+    subscription_id: str
+    customer_cancellation_reason: NotRequired[str]
+    customer_cancellation_comment: NotRequired[str]
+    canceled_at: str
+    ends_at: NotRequired[str]
+
+
+class SubscriptionCanceledEvent(Event):
+    if TYPE_CHECKING:
+        source: Mapped[Literal[EventSource.system]]
+        name: Mapped[Literal[SystemEvent.subscription_canceled]]
+        user_metadata: Mapped[SubscriptionCanceledMetadata]  # type: ignore[assignment]
+
+
 class SubscriptionCycledMetadata(TypedDict):
     subscription_id: str
```

Comment:
Could it be worth to add more metadata (amount, currency...) to this one as well?

Reply:
Yes!